lib/pbench/cli/agent/commands/generate_token.py

```diff
@@ -38,7 +38,11 @@ def execute(self):
             payload = {"message": response.text}
 
         if response.ok and "auth_token" in payload:
-            click.echo(payload["auth_token"])
+            if not self.context.output:
+                click.echo(payload["auth_token"])
+            else:
+                with open(self.context.output, "w") as ofp:
+                    ofp.write(f"{payload['auth_token']}\n")
```

Comment:
Alternately,
```
                Path(self.context.output).write_text(payload['auth_token'] + '\n')
```
(Since you used `Path.read_text()` in the test....)